builtin/logical/transit/path_decrypt.go

```diff
@@ -84,24 +153,43 @@ func (b *backend) pathDecryptWrite(
 		return logical.ErrorResponse("policy not found"), logical.ErrInvalidRequest
 	}
 
-	plaintext, err := p.Decrypt(context, nonce, ciphertext)
-	if err != nil {
-		switch err.(type) {
-		case errutil.UserError:
-			return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
-		case errutil.InternalError:
-			return nil, err
-		default:
-			return nil, err
+	for i, item := range batchInputItems {
+		if batchResponseItems[i].Error != "" {
+			continue
+		}
+
+		plaintext, err := p.Decrypt(item.DecodedContext, item.DecodedNonce, item.Ciphertext)
+		if err != nil {
+			switch err.(type) {
+			case errutil.UserError:
+				return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
```

Comment:
Seems like here it should set `batchResponseItems[i].Error` since it's a user error.